Update math functions

neon60 opened the pull request:
This PR focusing on math functions update and those changes are in kernel_language.rst.

I based my changes on PR: #3474 This will be merged in soon at Gerrit side.

neon60 committed:
Function should be code in the rst table

neon60 committed:
Fix single and double precision intrinsic functions

neon60 committed:
OCML_BASIC_ROUNDED_OPERATIONS define macro is undefined by default

neon60 committed:
Update integer intrinsic math functions

randyh62 reviewed (dismissed):
Seems like some of the images in common between understand/hardware_implementation and understand/programming_model could be consolidated to use a single reference? It seems like this is being done in the kernel_language.rst file.

neon60 commented:
> Seems like some of the images in common between understand/hardware_implementation and understand/programming_model could be consolidated to use a single reference? It seems like this is being done in the kernel_language.rst file.

The develop branch is slightly behind, this PR changes are focusing on kernel_language.rst.

Because of the data structure contribution rule. We have file duplication, or we do not state how to handle those.

matyas-streamhpc commented on docs/reference/kernel_language.rst:
```diff
@@ -327,441 +327,441 @@ Following is the list of supported single precision mathematical functions.
+    * - | ``void sincosf(float  x, float* sptr, float* cptr)``
         | Calculate the sine and cosine of the first input argument.
       - ✓
       - ✗ 
```
Can you check whether it is supported now?

It seems from this page that it is supported now, but I did not check it.
https://cgmb-hip.readthedocs.io/en/latest/autotoc_md76.html

Thanks!

neon60 replied:
Supported.

neon60 replied:
I have to check this, thanks for the heads up.

neon60 replied:
Fixed.

matyas-streamhpc commented on docs/reference/kernel_language.rst:
```diff
@@ -1372,255 +1462,255 @@ HIP supports the following atomic operations.
       - ✓
       - ✗
 
-    * - float safeAtomicAdd(float* address, float val)
```
missing apostrophes:
` * - ``float safeAtomicAdd(float* address, float val)`` `

neon60 replied:
Fixed. Thanks

neon60 committed:
PR findings

- Remove function duplication and rename math function variables
- Added missing math functions and updated description
- Minor fixes
